packages/orb-sync-lib/src/database/postgres.ts

```diff
@@ -23,18 +23,47 @@ export class PostgresClient {
 
     // Max 5 in parallel to avoid exhausting connection pool
     const chunkSize = 5;
+    const sqlQuery = this.constructUpsertSql(this.config.schema, table, tableSchema);
     const results: pg.QueryResult<T>[] = [];
 
     for (let i = 0; i < entries.length; i += chunkSize) {
+      const queries: Promise<pg.QueryResult<T>>[] = [];
       const chunk = entries.slice(i, i + chunkSize);
+      chunk.forEach((entry) => {
+        // Inject the values
+        const cleansed = this.cleanseArrayField(entry, tableSchema);
+        const prepared = sql(sqlQuery, {
+          useNullForMissing: true,
+        })(cleansed);
+
+        queries.push(this.pool.query(prepared.text, prepared.values));
+      });
 
+      results.push(...(await Promise.all(queries)));
+    }
+
+    return results.flatMap((it) => it.rows);
+  }
+
+  async updateMany<
+    T extends {
+      [Key: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    },
+  >(entries: T[], table: string, tableSchema: JsonSchema): Promise<T[]> {
+    if (!entries.length) return [];
+
+    // Max 5 in parallel to avoid exhausting connection pool
+    const chunkSize = 5;
```

Comment:
I think it would be better if we did one query (just Concat with semicolon) per batch instead of parallelizing them and doing so many individual one off queries

Reply:
You mean one single query that concatenates the update queries for (in this case) all 100 subscriptions or still parallelize things and have e.g. 2 parallel connections with one query each that concatenates 50 update queries?

Reply:
Goal is to reduce the number of connections being blocked. So ie chunk these into batches of 20 and then do one single concatenated update call, nothing in parallel

Reply:
The CRON is not supposed to take up all the connections that could otherwise block the regular webhooks

Reply:
Had a look into this. Turns out that Postgres does not allow prepared statements when using a batch of semicolon-separated SQL statements.

We can probably sacrifice the performance benefit.

The more import aspect is SQL-injection. It's difficult to overlook everything that could potentially happe. Not sure though what's even possible here since the data is going through the Supabase and Orb systems before. At least for subscriptions we would be fine (for now) because we don't save any data that stems from user input. Or do we have a "we always use prepared statements, no matter what" approach?

Reply:
nah it's fine, we can also just add a very basic escape function (i.e. if date arg contains ' or ", throw)